core/src/actor.rs

```diff
@@ -9,26 +9,79 @@ use bitcoin::{
     Address, TapSighash, TapTweakHash,
 };
 use bitcoin::{TapLeafHash, TapNodeHash, TapSighashType, TxOut};
+use bitvm::signatures::winternitz;
+
+/// Available transaction types for [`WinternitzDerivationPath`].
+#[derive(Clone, Copy, Debug)]
+pub enum TxType {
+    TimeTx,
+    KickoffTx,
+}
+
+/// Derivation path specification for Winternitz one time public key generation.
+#[derive(Debug, Clone, Copy)]
+pub struct WinternitzDerivationPath {
+    pub message_length: u32,
+    pub log_d: u32,
+    pub index: u32,
+    pub tx_type: TxType,
+    pub operator_idx: u32,
+    pub watchtower_idx: u32,
+    pub time_tx_idx: u32,
+}
+impl WinternitzDerivationPath {
+    fn to_vec(self) -> Vec<u8> {
+        [
+            vec![self.tx_type as u8],
+            [
+                self.operator_idx.to_be_bytes(),
+                self.watchtower_idx.to_be_bytes(),
+                self.time_tx_idx.to_be_bytes(),
+            ]
+            .concat(),
+        ]
+        .concat()
+    }
+}
+impl Default for WinternitzDerivationPath {
+    fn default() -> Self {
+        Self {
+            message_length: Default::default(),
+            log_d: 4,
+            index: Default::default(),
+            tx_type: TxType::TimeTx,
+            operator_idx: Default::default(),
+            watchtower_idx: Default::default(),
+            time_tx_idx: Default::default(),
+        }
+    }
+}
 
 #[derive(Debug, Clone)]
 pub struct Actor {
     pub keypair: Keypair,
     _secret_key: SecretKey,
+    winternitz_secret_key: Option<String>,
```

Comment:
This should be 32 bytes

Reply:
Fixed in https://github.com/chainwayxyz/clementine/pull/371/commits/aed085261ea75ee760f5ed0ae39ccbfe90a55140.